process/smartContract/hooks/vmAccountsDB.go

```diff
@@ -2,45 +2,88 @@ package hooks
 
 import (
 	"encoding/binary"
+	"github.com/ElrondNetwork/elrond-go/core"
 	"math/big"
 	"sync"
 
+	"github.com/ElrondNetwork/elrond-go/data"
 	"github.com/ElrondNetwork/elrond-go/data/state"
+	"github.com/ElrondNetwork/elrond-go/data/typeConverters"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
 	"github.com/ElrondNetwork/elrond-go/hashing/keccak"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/sharding"
 )
 
 // VMAccountsDB is a wrapper over AccountsAdapter that satisfy vmcommon.BlockchainHook interface
 type VMAccountsDB struct {
-	accounts state.AccountsAdapter
-	addrConv state.AddressConverter
+	accounts         state.AccountsAdapter
+	addrConv         state.AddressConverter
+	storageService   dataRetriever.StorageService
+	blockChain       data.ChainHandler
```

Comment:
Why does a VMAccountsDB need a blockchain and shardCoordinator?

Is VMAccountsDB more than a DB for the VM? Maybe change the name if it is more.

Reply:
changed